manta-pay/src/simulation/ledger/mod.rs

```diff
@@ -222,34 +222,57 @@ impl Ledger {
     }
 }
 
+/// Transfer Ledger Error
+#[cfg_attr(
+    feature = "serde",
+    derive(Deserialize, Serialize),
+    serde(crate = "manta_util::serde", deny_unknown_fields)
+)]
+#[derive(derivative::Derivative)]
+#[derivative(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum LedgerError {
+    /// Sender Ledger Error
+    SenderLedgerError,
+
+    /// Receiver Ledger Error
+    ReceiverLedgerError,
+
+    /// Transfer Ledger Error
+    TransferLedgerError,
+}
+
 impl SenderLedger<Parameters> for Ledger {
     type ValidNullifier = Wrap<Nullifier>;
     type ValidUtxoAccumulatorOutput = Wrap<UtxoAccumulatorOutput<Config>>;
     type SuperPostingKey = (Wrap<()>, ());
+    type Error = LedgerError;
 
     #[inline]
-    fn is_unspent(&self, nullifier: Nullifier) -> Option<Self::ValidNullifier> {
+    fn is_unspent(
+        &self,
+        nullifier: Nullifier,
+    ) -> Result<Option<Self::ValidNullifier>, Self::Error> {
         if self.nullifiers.contains(&nullifier) {
-            None
+            Ok(None)
```

Comment:
Shouldn't these Ok(None) be an error ?

Reply:
It's a reasonable suggestion. I've chosen to keep the changes minimal, i.e., when we returned `None` before now we'll return `Ok(None)` and when we had an `expect` now we'll return an error type. Feel free to change the error types as you want if you think they'd be more appropriate.